src/NLog/Targets/FileTarget.cs

```diff
@@ -1178,13 +1185,17 @@ private void DeleteOldDateArchive(string pattern)
                     }
                 }
 
+                /* TODO: The following block could use EnsureArchiveCount, but the behavior is not exactly the same.
+                 * The number of files to delete to "make room" is calculated using 'files.Count' instead of 'filesByDate.Count',
+                 * which I suspect to be a bug, since 'files' can contain filenames that are not actual archives. */
```

Comment:
I'd like to create a dedicated PR for this as soon as this one gets merged. Don't you prefer ?